### PR TITLE
Refactored Vungle Android adapter to separate Banner implementation

### PR DESCRIPTION
**Pull request by joshuajplagonera**
Refactored Vungle Android adapter to separate Banner implementation

### CODE REVIEW

File: ThirdPartyAdapters/vungle/vungle/src/main/java/com/vungle/mediation/VungleInterstitialAdapter.java [hunk 1]
```diff
@@ -192,217 +184,70 @@ void onAdFail(String placement) {
...
+            vungleBannerAdapter.detach();
           }
         };
     int adLayoutHeight = adSize.getHeightInPixels(context);
     // If the height is 0 (e.g. for inline adaptive banner requests), use the closest supported size
     // as the height of the adLayout wrapper.
     if (adLayoutHeight <= 0) {
       float density = context.getResources().getDisplayMetrics().density;
-      adLayoutHeight = Math.round(adConfig.getAdSize().getHeight() * density);
+      adLayoutHeight = Math.round(mAdConfig.getAdSize().getHeight() * density);
```
Comment (HuFangshuai) [inline ThirdPartyAdapters/vungle/vungle/src/main/java/com/vungle/mediation/VungleInterstitialAdapter.java:238]: Is using `mAdConfig` here make sense?

File: ThirdPartyAdapters/vungle/vungle/src/main/java/com/vungle/mediation/VungleInterstitialAdapter.java [hunk 2]
```diff
@@ -192,217 +184,70 @@ void onAdFail(String placement) {
...
+    // for its dimensions.
+    bannerAdContainer =
         new RelativeLayout(context) {
           @Override
           protected void onAttachedToWindow() {
             super.onAttachedToWindow();
-            if (mBannerRequest != null) {
-              mBannerRequest.attach();
-            }
+            vungleBannerAdapter.attach(this);
```
Comment (HuFangshuai) [inline ThirdPartyAdapters/vungle/vungle/src/main/java/com/vungle/mediation/VungleInterstitialAdapter.java:224]: Better to protected with null value check here. Because this method called in onAttachedToWindow callback and also can be set to a null value in onDestroy() method, I am not sure if there are any cases that can cause NPE.

File: ThirdPartyAdapters/vungle/vungle/src/main/java/com/vungle/mediation/VungleInterstitialAdapter.java [hunk 3]
```diff
@@ -192,217 +184,70 @@ void onAdFail(String placement) {
...
           }
 
           @Override
           protected void onDetachedFromWindow() {
             super.onDetachedFromWindow();
-            if (mBannerRequest != null) {
-              mBannerRequest.detach();
-            }
+            vungleBannerAdapter.detach();
```
Comment (HuFangshuai) [inline ThirdPartyAdapters/vungle/vungle/src/main/java/com/vungle/mediation/VungleInterstitialAdapter.java:230]: Same to onAttachedToWindow comment.

**Comment by HuFangshuai** [comment]
@ericleich Did you push new commit? There's no new update since my last comment.

**Comment by ericleich** [comment]
@HuFangshuai sorry about that, thought I pushed before. Should be out now, please check again. I confirmed that `vungleBannerAdapter.detach();` is now backed by a null check.

Also, due to the way I'm using [Copybara](https://github.com/google/copybara) to push this externally from our internal review, it appears to override the commit instead of a new one. That might make reviewing the diff since last time infeasible, sorry about that. I'll look into if there's a better way to make a new commit instead of override so that future changes have a better workflow.

**Comment by HuFangshuai** [comment]
Thanks, @ericleich. The changes look good to me.
